msmarco-passage-elser-throughput/challenges/default.json

```diff
@@ -0,0 +1,108 @@
+{
+  "name": "msmarco-passage-elser-throughput",
+  "description": "Test the throughput of ELSER in various scenarios",
+  "default": true,
+  "schedule": [
+    {
+      "operation": "delete-index"
+    },
+    {
+      "operation": {
+        "include-in-reporting": false,
+        "operation-type": "create-index"
+      }
+    },
+    {
+      "name": "create elser model and deployment",
+      "operation": {
+        "number_of_allocations": {{ number_of_allocations|default(1) }},
+        "threads_per_allocation": {{ threads_per_allocation|default(2) }},
+        "queue_capacity": {{ queue_capacity|default(1024) }},
+        "operation-type": "create-elser-model",
+        "include-in-reporting": false
+      }
+    },
+    {
+      "name": "setup-elser-pipeline",
+      "operation": {
+        "operation-type": "put-pipeline",
+        "id": "elser2",
+        "body": {
+          "description": "Pipes documents to ELSER",
+          "processors": [
+            {
+              "inference": {
+                "model_id": ".elser_model_1",
+                "target_field": "ml",
+                "field_map": {
+                  "body": "text_field"
+                },
+                "inference_config": {
+                  "text_expansion": {
+                    "results_field": "tokens"
+                  }
+                }
+              }
+            }
+          ],
+          "on_failure": [
```

Comment:
In general we recommend using `on_failure` as best practice but in this case it would be best to allow the error to propagate back to the caller and fail the bulk upload. If the model deployment was down for some reason the benchmark would run suspiciously fast and the cause can only be found be searching for an error in the index.

Reply:
What you are saying about bulk running suspiciously fast in its failure mode is definitely true, but I tried removing `on_failure`, and it doesn't return an error. Do you know if there is something else I need to do to get it to throw an exception?

Reply:
either way, the final report shows the failure rate on bulk as 100% when its failing, so I don't think its a big deal either way.